onnxruntime/core/providers/cpu/tensor/transpose.cc

```diff
@@ -332,20 +334,20 @@ static void TranposeSingleAxisOutwards(const std::vector<size_t>& permutations,
     }
     default: {
       // we need to use memcpy for each block
-      std::vector<uint8_t*> writers;
-      writers.resize(num_writers);
-
       for (int64_t l = 0; l < num_loops; ++l) {
-        for (auto w = 0; w < num_writers; ++w) {
-          writers[w] = (output_data + (w * writes_per_writer_per_loop * bytes_per_write));
-        }
+        uint8_t* output_for_first_writer = output_data;
 
         for (auto wwpl = 0; wwpl < writes_per_writer_per_loop; ++wwpl) {
+          uint8_t* output_for_current_writer = output_for_first_writer;
```

Comment:
Can the typo in "TranposeSingleAxisOutwards" and "TranposeSingleAxisInwards" be fixed (tranpose->transpose)?

Reply:
SingleAxisTranspose references the problem of the std::vector allocation. Can that comment be updated to reflect the new logic?

Reply:
Fixed typo.
Removed comment - the new logic should be faster than calling ComputeOffset and IncrementIndex so I don't think the comment applies anymore.

Regarding can the default paths be hit. I expect we would for the single axis moving outwards as perms of {1, 0, 2} will hit it if the value of the last dim * bytes for the data type is > 64. 

Moving inwards is less likely as we arbitrarily prefer outwards over inwards, so if two axes are swapping (e.g. perms of {1, 0, 2} could be 1 moving in or 0 moving out) we'll choose TransposeSingleAxisOutwards. That means to hit the default in TransposeSingleAxisInwards you'd need an axis to move inwards by more than one, and one or more dimensions after that (e.g. something like {1, 2, 0, 3} for perms). Not sure how probable that combination of perms is across all the different types of models. Also not sure if it's more likely if you have rank 5 data (e.g. perms of {0, 2, 3, 1, 4}). 

Given that, is the risk of erroring out on the default path in TransposeSingleAxisInwards (for falling back to a less performant implementation) worth the binary size savings?

Reply:
I didn't follow all the logic through to see how it would be hit, if it is being hit, then I'm fine with it as is. I couldn't tell if that logic only existed to represent the generic type. And following it back is why I noticed the typo today. :)